pkg/cloudprovider/providers/azure/azure_loadbalancer.go

```diff
@@ -435,18 +442,20 @@ func (az *Cloud) cleanupLoadBalancer(clusterName string, service *v1.Service, is
 
 		// Public IP can be deleted after frontend ip configuration rule deleted.
 		if publicIPToCleanup != nil {
-			// Only delete an IP address if we created it, deducing by name.
+			// Only delete an IP address if we created it, deducing by name and resource group.
```

Comment:
Is the cleanup logic going to work correctly if a user _edits_ a service and changes which resource group its public IP is in?

Reply:
The existing code cannot handle it too. Mainly because we have no way to know which pip is actually created by k8s. I'm basically restricting the condition to delete public IP by matching name and **rg**. So in the worse case (the one you described), it didn't cleanup the pip which is not that bad I think. VS. accidentally deleted a pip in the provided rg. Thoughts?

Reply:
Maybe also add tags to k8s-managed public IPs?